php/php.dbgp/src/org/netbeans/modules/php/dbgp/breakpoints/BreakpointModel.java

```diff
@@ -119,15 +126,25 @@ public String getShortDescription(Object node) throws UnknownTypeException {
 
     public void setCurrentStack(Stack stack, DebugSession session) {
         if (stack == null) {
-            synchronized (myCurrentBreakpoints) {
-                AbstractBreakpoint breakpoint = myCurrentBreakpoints.remove(session);
-                fireChangeEvent(new ModelEvent.NodeChanged(this, breakpoint));
-            }
+            removeCurrentBreakpoint(session);
```

Comment:
Nitpick: This line can be moved inside the `synchronized` block above (it uses the same lock, so it would avoid the situation where we get the lock, release the lock and get the lock again).

Reply:
Sorry, I'm not quite sure what you mean.

Reply:
Sorry, my bad, the `synchronized` itself block was removed (and replaced by this call). I somehow missed it.